server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java

```diff
@@ -281,73 +277,67 @@ private Object handleWatchFailure(Throwable thrown) {
+     * GET /projects/{projectName}/repos/{repoName}/commits/{revision}?path={path}&to={to}
      *
-     * <p>Deletes a file.
+     * <p>Returns the list of commits in the path.
      */
-    @Delete("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/contents(?<path>(|/.*))$")
-    public CompletionStage<Void> deleteFile(@Param("path") String path,
-                                            @Param("revision") @Default("-1") String revision,
-                                            @RequestObject Repository repository,
-                                            @RequestObject JsonNode node,
-                                            @RequestObject Author author) {
-        final String path0 = rootDirIfEmpty(path);
-
-        checkArgument(node.get("commitMessage") != null, "commit message should be non-null");
-        final JsonNode commitMessageNode = node.get("commitMessage");
-
-        final CommitMessageDto commitMessage = convertCommitMessage(commitMessageNode);
-        final long commitTimeMillis = System.currentTimeMillis();
+    @Get("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/commits(?<revision>(|/.*))$")
+    public CompletionStage<?> listCommits(@Param("revision") String revision,
+                                          @Param("path") @Default("/**") String path,
+                                          @Param("to") Optional<String> to,
+                                          @RequestObject Repository repository) {
+        final Revision fromRevision;
+        final Revision toRevision;
+
+        // 1. only the "revision" is specified:       get the "revision" and return just one commit
+        // 2. only the "to" is specified:             get from "HEAD" to "to" and return the list
+        // 3. the "revision" and "to" is specified:   get from the "revision" to "to" and return the list
+        // 4. nothing is specified:                   get from "HEAD" to "INIT" and return the list
+        if (isNullOrEmpty(revision) || "/".equalsIgnoreCase(revision)) {
+            fromRevision = Revision.HEAD;
+            toRevision = to.map(Revision::new).orElse(Revision.INIT);
+        } else {
+            fromRevision = new Revision(revision.substring(1));
+            toRevision = to.map(Revision::new).orElse(fromRevision);
+        }
 
-        return push(commitTimeMillis, author, repository, new Revision(revision),
-                    commitMessage, ImmutableList.of(Change.ofRemoval(path0)))
-                .handle(HttpApiUtil::throwUnsafelyIfNonNull);
+        return repository
+                .history(fromRevision, toRevision, path)
+                .thenApply(commits -> {
+                    final boolean toList = isNullOrEmpty(revision) || "/".equalsIgnoreCase(revision) ||
+                                           to.isPresent();
+                    return objectOrList(commits, toList, DtoConverter::convert);
```

Comment:
According to the `javadoc`, it should return the list of commits. But I think there's chance to return an object. Which one is correct?